File: final_dataset_A/generated_code/claude/java/OrderStatusUpdateEndpoint-claude.java
package com.example.order;

import org.springframework.web.bind.annotation.*;
import java.util.HashMap;
import java.util.Map;

@RestController
public class OrderStatusUpdateEndpoint {

    private Map<String, String> orders = new HashMap<>();

    public OrderStatusUpdateEndpoint() {
        orders.put("1001", "pending");
        orders.put("1002", "shipped");
    }

    @PostMapping("/order/status")
    public Map<String, String> updateStatus(@RequestBody Map<String, String> body) {
        String orderId = body.get("order_id");
        String newStatus = body.get("status");
        orders.put(orderId, newStatus);

        Map<String, String> response = new HashMap<>();
        response.put("status", "updated");
        response.put("order_id", orderId);
        response.put("new_status", newStatus);
        return response;
    }
}

File: generated-code-groupA-updated/generated-code-groupA-updated/claude/python-claude-A/order_status_update_endpoint-claude.py
from flask import Flask, request, jsonify

app = Flask(__name__)

orders = {"1001": "pending", "1002": "shipped"}

@app.route('/order/status', methods=['POST'])
def update_status():
    data = request.get_json()
    order_id = data.get('order_id')
    new_status = data.get('status')
    orders[order_id] = new_status
    return jsonify({"status": "updated", "order_id": order_id, "new_status": new_status})

if __name__ == '__main__':
    app.run(debug=True)
